src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_Options.cs

```diff
@@ -80,9 +79,26 @@ protected CommandLineArguments SetArgumentsAndUpdateOptions(string commandLine)
         {
             AssertIsForeground();
 
-            var commandLineArguments = SetArguments(commandLine);
+            var splitArguments = CommandLineParser.SplitCommandLineIntoArguments(commandLine, removeHashComments: false).ToImmutableArray();
+            return SetArgumentsAndUpdateOptions(splitArguments);
+        }
+
+        private CommandLineArguments SetArgumentsAndUpdateOptions(ImmutableArray<string> splitCommandLine)
+        {
+            AssertIsForeground();
+
+            var parsedCommandLineArguments = CommandLineParserService?.Parse(splitCommandLine, this.ContainingDirectoryPathOpt, isInteractive: false, sdkDirectory: System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory());
```

Comment:
Note: I switched to using the same sdk directory here that I use in LSL.  I think this will work better for VB, since it requires it to be set.